File: Objetivo 3/Objetivo3_Exc_L1_PHP/model/Desenvolvedor.php
<?php

include_once "Funcionario.php";

/**
 * Description of Desenvolvedor
 *
 * @author vagner
 */
class Desenvolvedor extends Funcionario{
    
    /*
     * Onde você deve inserir as regras de negócio? Sem dúvidas, é no modelo. O controlador só deve gerenciar o fluxo entre a Model e a View.
     */
    public function getBonus() {
        return parent::getSalario() * 0.05;
    }
    
    public function __toString() {
        return "Desenvolvedor{" . parent::__toString() . "}";
    }

}

File: Objetivo 3/Objetivo3_Exc_L1_Java/src/model/Desenvolvedor.java
package model;

/**
 *
 * @author vagner
 */
public class Desenvolvedor extends Funcionario{

    @Override
    public double getBonus() {
        return super.getSalario() * 0.05;
    }

    @Override
    public String toString() {
        return "Desenvolvedor{" + super.toString() + "}";
    }
    
    

}
